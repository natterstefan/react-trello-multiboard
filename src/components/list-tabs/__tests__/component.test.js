import React from 'react'
import { shallow } from 'enzyme'
import { assign } from 'lodash'

import Tabs from '@material-ui/core/Tabs'
import Tab from '@material-ui/core/Tab'

import ListTabs from '../component'

const props = {
  classes: {},
  changeListToggle: jest.fn(),
  listsConfig: ['list-1', 'list-2'],
  toggleList: 'list-1',
}

describe('Component/ListTabs', () => {
  beforeEach(() => {
    props.changeListToggle.mockReset()
  })

  test('should render without throwing an error', () => {
    const wrapper = shallow(<ListTabs {...props} />).dive()
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find(Tabs).length).toEqual(1)
    expect(wrapper.find(Tab).length).toEqual(2)
  })

  test('should render null when no listsConfig is available', () => {
    const newProps = assign({}, props, { listsConfig: [] })
    expect(
      shallow(<ListTabs {...newProps} />)
        .dive()
        .html(),
    ).toEqual(null)
  })

  test('changes of a Tab should call handleChange', () => {
    const wrapper = shallow(<ListTabs {...props} />).dive()
    wrapper
      .find(Tabs)
      .first()
      .simulate('change', null, 0)
    expect(props.changeListToggle).toHaveBeenCalledTimes(1)
    expect(props.changeListToggle).toHaveBeenCalledWith('list-1')
  })

  test('should render empty html when no listsConfig is available', () => {
    const newProps = assign({}, props, { listsConfig: [] })
    expect(shallow(<ListTabs {...newProps} />).html()).toEqual('')
  })
})
