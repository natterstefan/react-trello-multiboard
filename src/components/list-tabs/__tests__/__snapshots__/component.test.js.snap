// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component/ListTabs should render without throwing an error 1`] = `
<div
  className="ListTabs-root-1"
>
  <WithStyles(AppBar)
    color="default"
    position="static"
  >
    <WithStyles(Tabs)
      indicatorColor="primary"
      onChange={[Function]}
      scrollButtons="on"
      scrollable={true}
      textColor="primary"
      value="list-1"
    >
      <WithStyles(Tab)
        disabled={true}
        key="0"
        label="list-1"
      />
      <WithStyles(Tab)
        disabled={false}
        key="1"
        label="list-2"
      />
    </WithStyles(Tabs)>
  </WithStyles(AppBar)>
</div>
`;
